plugins/inputs/mongodb/mongodb.go

```diff
@@ -132,6 +142,12 @@ func (m *MongoDB) Gather(acc telegraf.Accumulator) error {
 		wg.Add(1)
 		go func(srv *Server) {
 			defer wg.Done()
+			if m.IgnoreUnreachableHosts {
+				if err := srv.ping(); err != nil {
```

Comment:
Does pinging attempt to reconnect? In other words, if a server goes down, can it ever come back up or is it going to fail ping each gather from then on?

Reply:
`mongodb` go client spawns one go routine per conn and use it to monitor the connexion. Transient errors should be recoverable from the client it self.